src/components/ArticleById.jsx

```diff
@@ -1,42 +1,74 @@
-import { getArticleById } from "../api";
+import { getArticleById, updateVote } from "../api";
+import ArticleComments from "./ArticleComments";
 
 import React, { useState, useEffect } from "react";
 import { useParams } from "react-router-dom";
 
-import Navbar from "./Navbar";
 import { IconButton } from "@mui/material";
 import CommentTwoToneIcon from "@mui/icons-material/CommentTwoTone";
 import ThumbUpTwoToneIcon from "@mui/icons-material/ThumbUpTwoTone";
 
 const ArticleById = () => {
   const { article_id } = useParams();
-  const [article, setArticle] = useState([]);
+  const [article, setArticle] = useState({});
+  const [voteCounter, setVoteCounter] = useState();
+
+  const [disable, setDisable] = useState(false);
+  const [hasError, setHasError] = useState(false);
 
   useEffect(() => {
     getArticleById(article_id).then(res => {
       setArticle(res);
+      setVoteCounter(res.votes);
     });
   }, [article_id]);
 
+  console.log(article);
+  const handleVote = () => {
+    if (disable === false) {
+      updateVote(article_id, 1).catch(err => {
+        setHasError(true);
+        setVoteCounter(0);
+      });
+      setVoteCounter(currVoteCounter => currVoteCounter + 1);
+      setDisable(true);
+      console.log(disable);
+    }
+    if (disable === true) {
+      updateVote(article_id, -1).catch(err => {
+        setHasError(true);
+        setVoteCounter(0);
```

Comment:
This might be misleading to a user. If an article has several votes and there is an error, the user will then be presented with zero votes. Consider handling it in a way that removes one from the current amount of votes. Something like line 33 with a -1.

Reply:
Done.
Could you double check please?